src/m365/aad/commands/group/group-user-list.ts

```diff
@@ -0,0 +1,198 @@
+import { aadGroup } from '../../../../utils/aadGroup.js';
+import { formatting } from '../../../../utils/formatting.js';
+import { odata } from '../../../../utils/odata.js';
+import { validation } from '../../../../utils/validation.js';
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId?: string;
+  groupDisplayName?: string;
+  role?: string;
+  properties?: string;
+  filter?: string;
+}
+
+interface ExtendedUser extends User {
+  roles: string[];
+}
+
+class AadGroupUserListCommand extends GraphCommand {
+  public get name(): string {
+    return commands.GROUP_USER_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists users of a specific Azure AD group';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'userPrincipalName', 'roles'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupDisplayName: typeof args.options.groupDisplayName !== 'undefined',
+        role: typeof args.options.role !== 'undefined',
+        properties: typeof args.options.properties !== 'undefined',
+        filter: typeof args.options.filter !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "-i, --groupId [groupId]"
+      },
+      {
+        option: "-n, --groupDisplayName [groupDisplayName]"
+      },
+      {
+        option: "-r, --role [type]",
+        autocomplete: ["Owner", "Member"]
+      },
+      {
+        option: "-p, --properties [properties]"
+      },
+      {
+        option: "-f, --filter [filter]"
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['groupId', 'groupDisplayName']
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.role) {
+          if (['Owner', 'Member'].indexOf(args.options.role) === -1) {
+            return `${args.options.role} is not a valid role value. Allowed values Owner|Member`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const groupId = await this.getGroupId(args.options);
+
+      let users: User[] = [];
+
+      switch (args.options.role) {
```

Comment:
I find this switch statement rather verbose, especially combined with how we need to loop through the owners anyway. I'd like to suggest as alternative just using two if-statements. Along these lines or something like that:

```typescript
const users: ExtendedUser[] = [];

if (!args.options.role || args.options.role === 'Owner') {
  const owners = await this.getUsers(args.options, 'Owners', groupId, logger);
  owners.forEach(owner => users.push({ ...owner, roles: ['Owner'] }));        
}

if (!args.options.role || args.options.role === 'Member') {
  const members = await this.getUsers(args.options, 'Members', groupId, logger);
  
  members.forEach((member: ExtendedUser) => {
    const user = users.find((u: ExtendedUser) => u.id === member.id);

    if (user !== undefined) {
      user.roles.push('Member');
    }
    else {
      users.push(member);
    }
  });   
}
```

You can now also remove setting the owner role property within the `getUsers` function.